phpfpm input plugin panics in fcgi mode with json format

### Relevant telegraf.conf

```toml
[[inputs.phpfpm]]
  urls = ["fcgi://php-fpm:9001/status?full&json"]
  format = "json"
```


### Logs from Telegraf

```text
2023-12-20T12:29:49Z I! Loading config: /etc/telegraf/telegraf.conf
2023-12-20T12:29:49Z I! Starting Telegraf 1.29.1 brought to you by InfluxData the makers of InfluxDB
2023-12-20T12:29:49Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 60 outputs, 6 secret-stores
2023-12-20T12:29:49Z I! Loaded inputs: phpfpm
2023-12-20T12:29:49Z I! Loaded aggregators: 
2023-12-20T12:29:49Z I! Loaded processors: 
2023-12-20T12:29:49Z I! Loaded secretstores: 
2023-12-20T12:29:49Z I! Loaded outputs: prometheus_client
2023-12-20T12:29:49Z I! Tags enabled: host=e42eae5ec81c
2023-12-20T12:29:49Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"e42eae5ec81c", Flush Interval:10s
2023-12-20T12:29:49Z D! [agent] Initializing plugins
2023-12-20T12:29:49Z D! [agent] Connecting outputs
2023-12-20T12:29:49Z D! [agent] Attempting connection to [outputs.prometheus_client]
2023-12-20T12:29:49Z I! [outputs.prometheus_client] Listening on http://[::]:9273/metrics
2023-12-20T12:29:49Z D! [agent] Successfully connected to outputs.prometheus_client
2023-12-20T12:29:49Z D! [agent] Starting service inputs
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x3392cd1]
```


### System info

Telegraf 1.29.1, Ubuntu 22.04, Docker 24.0.7, Docker Compose 2.21.0

### Docker

```yaml
# docker-compose.yml
services:
  php-fpm:
    image: docker.io/library/php:8.3.0-fpm-alpine3.19
    volumes:
      - ./zz-php-fpm.conf:/usr/local/etc/php-fpm.d/zz-php-fpm.conf:ro
  telegraf83:
    image: docker.io/library/telegraf:1.29.1-alpine
    entrypoint:
      - /entrypoint.sh
    command:
      - telegraf
      - --debug
    volumes:
      - ./telegraf.conf:/etc/telegraf/telegraf.conf:ro
```

```ini
# zz-php-fpm.conf
[www]
pm.status_listen = 9001
pm.status_path = /status
```

```toml
[[inputs.phpfpm]]
  urls = ["fcgi://php-fpm:9001/status?full&json"]
  format = "json"
```

### Steps to reproduce

1. `docker compose up --detach`
2. `docker compose logs --follow telegraf`


### Expected behavior

No panic.

### Actual behavior

Panic.

### Additional info

When I add a Nginx FastCGI proxy between Telegraf and PHP-FPM and configure Telegraf to use HTTP to get the metrics there is no panic (see log output below).

I guess one issue is that the [gatherFcgi](https://github.com/influxdata/telegraf/blob/9a469977dd46c73a91dfaf73dd306d6a7dd1ec7a/plugins/inputs/phpfpm/phpfpm.go#L193) method is not setting the QUERY_STRING (`full&json` in my case). It might also be necessary to remove the query (`?full&json`) from the `statusPath` when setting `SCRIPT_NAME` and `SCRIPT_FILENAME`.

```
2023-12-20T12:29:50Z I! Loading config: /etc/telegraf/telegraf.conf
2023-12-20T12:29:50Z I! Starting Telegraf 1.29.1 brought to you by InfluxData the makers of InfluxDB
2023-12-20T12:29:50Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 60 outputs, 6 secret-stores
2023-12-20T12:29:50Z I! Loaded inputs: phpfpm
2023-12-20T12:29:50Z I! Loaded aggregators: 
2023-12-20T12:29:50Z I! Loaded processors: 
2023-12-20T12:29:50Z I! Loaded secretstores: 
2023-12-20T12:29:50Z I! Loaded outputs: prometheus_client
2023-12-20T12:29:50Z I! Tags enabled: host=c72f20970a05
2023-12-20T12:29:50Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"c72f20970a05", Flush Interval:10s
2023-12-20T12:29:50Z I! [outputs.prometheus_client] Listening on http://[::]:9273/metrics
``

--- a/plugins/inputs/phpfpm/phpfpm.go
+++ b/plugins/inputs/phpfpm/phpfpm.go
@@ -83,7 +83,7 @@ type phpfpm struct {
 
 	tls.ClientConfig
 	client *http.Client
-	log    telegraf.Logger
+	Log    telegraf.Logger
 }
 
 func (*phpfpm) SampleConfig() string {
@@ -300,7 +300,7 @@ func parseLines(r io.Reader, acc telegraf.Accumulator, addr string) {
 func (p *phpfpm) parseJSON(r io.Reader, acc telegraf.Accumulator, addr string) {
 	var metrics JSONMetrics
 	if err := json.NewDecoder(r).Decode(&metrics); err != nil {
-		p.log.Errorf("Unable to decode JSON response: %s", err)
+		p.Log.Errorf("Unable to decode JSON response: %s", err)
 		return
 	}
 	timestamp := time.Now()

--- a/plugins/inputs/phpfpm/phpfpm_test.go
+++ b/plugins/inputs/phpfpm/phpfpm_test.go
@@ -6,9 +6,12 @@
 package phpfpm
 
 import (
+	"bytes"
 	"crypto/rand"
+	_ "embed"
 	"encoding/binary"
 	"fmt"
+	"log"
 	"net"
 	"net/http"
 	"net/http/fcgi"
@@ -77,8 +80,6 @@ func TestPhpFpmGeneratesMetrics_From_Http(t *testing.T) {
 }
 
 func TestPhpFpmGeneratesJSONMetrics_From_Http(t *testing.T) {
-	outputSampleJSON, err := os.ReadFile("testdata/phpfpm.json")
-	require.NoError(t, err)
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Content-Type", "text/json")
 		w.Header().Set("Content-Length", strconv.Itoa(len(outputSampleJSON)))
@@ -360,7 +361,10 @@ max children reached: 2
 slow requests:        1
 `
 
-func TestPhpFpmLogWithoutPanic(t *testing.T) {
+//go:embed testdata/phpfpm.json
+var outputSampleJSON []byte
+
+func TestPhpFpmParseJSON_Log_Error_Without_Panic_When_When_JSON_Is_Invalid(t *testing.T) {
 	p := &phpfpm{}
 	// AddInput sets the Logger
 	if err := shim.New().AddInput(p); err != nil {
@@ -368,5 +372,20 @@ func TestPhpFpmLogWithoutPanic(t *testing.T) {
 		return
 	}
 
-	p.Log.Debug("test")
+	// capture log output
+	var logOutput bytes.Buffer
+	log.SetOutput(&logOutput)
+	defer func() {
+		log.SetOutput(os.Stderr)
+	}()
+
+	// parse valid JSON without panic and without log output
+	validJSON := outputSampleJSON
+	require.NotPanics(t, func() { p.parseJSON(bytes.NewReader(validJSON), &testutil.NopAccumulator{}, "") })
+	require.Equal(t, logOutput.String(), "")
+
+	// parse invalid JSON without panic but with log output
+	invalidJSON := []byte("X")
+	require.NotPanics(t, func() { p.parseJSON(bytes.NewReader(invalidJSON), &testutil.NopAccumulator{}, "") })
+	require.Contains(t, logOutput.String(), "E! Unable to decode JSON response: invalid character 'X' looking for beginning of value")
 }

--- a/plugins/inputs/phpfpm/phpfpm_test.go
+++ b/plugins/inputs/phpfpm/phpfpm_test.go
@@ -19,6 +19,7 @@ import (
 
 	"github.com/stretchr/testify/require"
 
+	"github.com/influxdata/telegraf/plugins/common/shim"
 	"github.com/influxdata/telegraf/plugins/parsers/influx"
 	"github.com/influxdata/telegraf/testutil"
 )
@@ -94,7 +95,7 @@ func TestPhpFpmGeneratesJSONMetrics_From_Http(t *testing.T) {
 	input := &phpfpm{
 		Urls:   []string{server.URL + "?full&json"},
 		Format: "json",
-		log:    testutil.Logger{},
+		Log:    testutil.Logger{},
 	}
 	require.NoError(t, input.Init())
 
@@ -358,3 +359,14 @@ max active processes: 1
 max children reached: 2
 slow requests:        1
 `
+
+func TestPhpFpmLogWithoutPanic(t *testing.T) {
+	p := &phpfpm{}
+	// AddInput sets the Logger
+	if err := shim.New().AddInput(p); err != nil {
+		t.Error(err)
+		return
+	}
+
+	p.Log.Debug("test")
+}

--- a/plugins/inputs/phpfpm/phpfpm_test.go
+++ b/plugins/inputs/phpfpm/phpfpm_test.go
@@ -382,7 +382,7 @@ func TestPhpFpmParseJSON_Log_Error_Without_Panic_When_When_JSON_Is_Invalid(t *te
 	// parse valid JSON without panic and without log output
 	validJSON := outputSampleJSON
 	require.NotPanics(t, func() { p.parseJSON(bytes.NewReader(validJSON), &testutil.NopAccumulator{}, "") })
-	require.Equal(t, logOutput.String(), "")
+	require.Equal(t, "", logOutput.String())
 
 	// parse invalid JSON without panic but with log output
 	invalidJSON := []byte("X")

